<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Serial</class>
 <widget class="QMainWindow" name="Serial">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>783</width>
    <height>581</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Serial</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="8" column="8">
     <widget class="QLineEdit" name="pos_ang">
      <property name="text">
       <string>0</string>
      </property>
     </widget>
    </item>
    <item row="6" column="6">
     <widget class="QLabel" name="label">
      <property name="styleSheet">
       <string notr="true">font: 12pt &quot;Adobe Arabic&quot;;
font: 18pt &quot;Adobe Arabic&quot;;</string>
      </property>
      <property name="text">
       <string>       x:</string>
      </property>
     </widget>
    </item>
    <item row="0" column="3">
     <widget class="QPushButton" name="home">
      <property name="text">
       <string>home</string>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QPushButton" name="bt_disconect">
      <property name="text">
       <string>断开连接</string>
      </property>
     </widget>
    </item>
    <item row="15" column="0" colspan="10">
     <widget class="QTextBrowser" name="message"/>
    </item>
    <item row="0" column="0">
     <widget class="QLineEdit" name="serial">
      <property name="text">
       <string>COM</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QPushButton" name="bt_connect">
      <property name="text">
       <string>连接</string>
      </property>
     </widget>
    </item>
    <item row="1" column="2">
     <widget class="QPushButton" name="pos_save">
      <property name="text">
       <string>pos save</string>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QPushButton" name="sand">
      <property name="text">
       <string>发送</string>
      </property>
     </widget>
    </item>
    <item row="1" column="3">
     <widget class="QPushButton" name="pushButton">
      <property name="text">
       <string>home adj</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLineEdit" name="et_send"/>
    </item>
    <item row="1" column="8">
     <widget class="QLabel" name="switch_side">
      <property name="text">
       <string>红场</string>
      </property>
     </widget>
    </item>
    <item row="1" column="6">
     <widget class="QLabel" name="label_3">
      <property name="text">
       <string>现在状态：</string>
      </property>
     </widget>
    </item>
    <item row="0" column="8">
     <widget class="QPushButton" name="switch_blue">
      <property name="text">
       <string>蓝场</string>
      </property>
     </widget>
    </item>
    <item row="0" column="6">
     <widget class="QPushButton" name="switch_red">
      <property name="text">
       <string>红场</string>
      </property>
     </widget>
    </item>
    <item row="3" column="8">
     <widget class="QPushButton" name="toPos">
      <property name="text">
       <string>to_pos</string>
      </property>
     </widget>
    </item>
    <item row="2" column="6">
     <widget class="QPushButton" name="debug">
      <property name="text">
       <string>debug</string>
      </property>
     </widget>
    </item>
    <item row="2" column="8">
     <widget class="QPushButton" name="debug_print">
      <property name="text">
       <string>debug_print</string>
      </property>
     </widget>
    </item>
    <item row="7" column="6">
     <widget class="QLabel" name="label_2">
      <property name="styleSheet">
       <string notr="true">font: 18pt &quot;Adobe Arabic&quot;;</string>
      </property>
      <property name="text">
       <string>       y:</string>
      </property>
     </widget>
    </item>
    <item row="6" column="8">
     <widget class="QLineEdit" name="pos_x">
      <property name="text">
       <string>0</string>
      </property>
     </widget>
    </item>
    <item row="9" column="8">
     <widget class="QPushButton" name="go">
      <property name="text">
       <string>go</string>
      </property>
     </widget>
    </item>
    <item row="4" column="6">
     <widget class="QLabel" name="y">
      <property name="text">
       <string>y:</string>
      </property>
     </widget>
    </item>
    <item row="5" column="8">
     <widget class="QPushButton" name="toStart">
      <property name="text">
       <string>to_start</string>
      </property>
     </widget>
    </item>
    <item row="2" column="0" rowspan="13" colspan="4">
     <widget class="QTabWidget" name="tabWidget">
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="launch">
       <attribute name="title">
        <string>发射</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="3">
         <widget class="QPushButton" name="target">
          <property name="text">
           <string>target</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="et_select">
          <property name="text">
           <string>1</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QPushButton" name="posSelect">
          <property name="text">
           <string>pos select</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="printLeft">
          <property name="text">
           <string>printLeft</string>
          </property>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QLineEdit" name="target_l"/>
        </item>
        <item row="5" column="0">
         <widget class="QPushButton" name="left_target">
          <property name="text">
           <string>选择发射台</string>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QPushButton" name="printRight">
          <property name="text">
           <string>printRight</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QPushButton" name="yawLeft">
          <property name="text">
           <string>高yaw</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QLineEdit" name="left_speed">
          <property name="text">
           <string>7.7</string>
          </property>
         </widget>
        </item>
        <item row="4" column="3">
         <widget class="QPushButton" name="speedRight">
          <property name="text">
           <string>低speed</string>
          </property>
         </widget>
        </item>
        <item row="1" column="4">
         <widget class="QLineEdit" name="right_no">
          <property name="text">
           <string>1</string>
          </property>
         </widget>
        </item>
        <item row="3" column="4">
         <widget class="QLineEdit" name="right_yaw">
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
        <item row="2" column="3">
         <widget class="QPushButton" name="pitchRight">
          <property name="text">
           <string>低pitch</string>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QPushButton" name="speedLeft">
          <property name="text">
           <string>高speed</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QPushButton" name="pitchLeft">
          <property name="text">
           <string>高pitch</string>
          </property>
         </widget>
        </item>
        <item row="2" column="4">
         <widget class="QLineEdit" name="right_pitch">
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
        <item row="4" column="4">
         <widget class="QLineEdit" name="right_speed">
          <property name="text">
           <string>7.7</string>
          </property>
         </widget>
        </item>
        <item row="3" column="3">
         <widget class="QPushButton" name="yawRight">
          <property name="text">
           <string>低yaw</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLineEdit" name="left_pitch">
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
        <item row="0" column="4">
         <widget class="QLineEdit" name="et_target">
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="left_no">
          <property name="text">
           <string>1</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QLineEdit" name="left_yaw">
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
        <item row="6" column="1">
         <widget class="QPushButton" name="left_modi">
          <property name="text">
           <string>左modi</string>
          </property>
         </widget>
        </item>
        <item row="5" column="4">
         <widget class="QLineEdit" name="target_r"/>
        </item>
        <item row="7" column="0">
         <widget class="QPushButton" name="nextLeft">
          <property name="text">
           <string>左下一个台</string>
          </property>
         </widget>
        </item>
        <item row="8" column="0">
         <widget class="QPushButton" name="left_next">
          <property name="text">
           <string>左继续</string>
          </property>
         </widget>
        </item>
        <item row="7" column="4">
         <widget class="QPushButton" name="right_del">
          <property name="text">
           <string>右删除</string>
          </property>
         </widget>
        </item>
        <item row="8" column="3">
         <widget class="QPushButton" name="right_next">
          <property name="text">
           <string>右继续</string>
          </property>
         </widget>
        </item>
        <item row="7" column="1">
         <widget class="QPushButton" name="left_del">
          <property name="text">
           <string>左删除</string>
          </property>
         </widget>
        </item>
        <item row="8" column="4">
         <widget class="QPushButton" name="right_fly">
          <property name="text">
           <string>右发射</string>
          </property>
          <property name="shortcut">
           <string>Ctrl+*</string>
          </property>
         </widget>
        </item>
        <item row="6" column="3">
         <widget class="QPushButton" name="right_add">
          <property name="text">
           <string>右add</string>
          </property>
         </widget>
        </item>
        <item row="6" column="0">
         <widget class="QPushButton" name="left_add">
          <property name="text">
           <string>左add</string>
          </property>
         </widget>
        </item>
        <item row="8" column="1">
         <widget class="QPushButton" name="left_fly">
          <property name="text">
           <string>左发射</string>
          </property>
          <property name="shortcut">
           <string>Ctrl+/</string>
          </property>
         </widget>
        </item>
        <item row="5" column="3">
         <widget class="QPushButton" name="right_target">
          <property name="text">
           <string>选择发射台</string>
          </property>
         </widget>
        </item>
        <item row="7" column="3">
         <widget class="QPushButton" name="nextRight">
          <property name="text">
           <string>右下一个台</string>
          </property>
         </widget>
        </item>
        <item row="6" column="4">
         <widget class="QPushButton" name="right_modi">
          <property name="text">
           <string>右modi</string>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QPushButton" name="left_pitch_bt">
          <property name="minimumSize">
           <size>
            <width>32</width>
            <height>0</height>
           </size>
          </property>
          <property name="text">
           <string>0.000</string>
          </property>
         </widget>
        </item>
        <item row="3" column="2">
         <widget class="QPushButton" name="left_yaw_bt">
          <property name="text">
           <string>0.000</string>
          </property>
         </widget>
        </item>
        <item row="4" column="2">
         <widget class="QPushButton" name="left_speed_bt">
          <property name="text">
           <string>7.700</string>
          </property>
         </widget>
        </item>
        <item row="2" column="5">
         <widget class="QPushButton" name="right_pitch_bt">
          <property name="text">
           <string>0.000</string>
          </property>
         </widget>
        </item>
        <item row="3" column="5">
         <widget class="QPushButton" name="right_yaw_bt">
          <property name="text">
           <string>0.000</string>
          </property>
         </widget>
        </item>
        <item row="4" column="5">
         <widget class="QPushButton" name="right_speed_bt">
          <property name="text">
           <string>7.700</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabWidgetPage2">
       <attribute name="title">
        <string>位置</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="8" column="1">
         <widget class="QLineEdit" name="load_ang">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item row="8" column="0">
         <widget class="QLabel" name="label_19">
          <property name="text">
           <string>loadang:</string>
          </property>
         </widget>
        </item>
        <item row="5" column="2">
         <widget class="QPushButton" name="pushButton_2">
          <property name="text">
           <string>改</string>
          </property>
         </widget>
        </item>
        <item row="10" column="1">
         <widget class="QLineEdit" name="modi_y">
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QLineEdit" name="lineEdit">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item row="9" column="2">
         <widget class="QLabel" name="label_23">
          <property name="text">
           <string>ang</string>
          </property>
         </widget>
        </item>
        <item row="4" column="2">
         <widget class="QPushButton" name="posPrint">
          <property name="text">
           <string>pos print</string>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QPushButton" name="auto_print">
          <property name="text">
           <string>auto print</string>
          </property>
         </widget>
        </item>
        <item row="8" column="2">
         <widget class="QPushButton" name="load">
          <property name="text">
           <string>改</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLabel" name="label_15">
          <property name="text">
           <string>nowy:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_13">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>nowx:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QLabel" name="label_14">
          <property name="text">
           <string>nowang:</string>
          </property>
         </widget>
        </item>
        <item row="10" column="0">
         <widget class="QLineEdit" name="modi_x">
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
        <item row="9" column="0">
         <widget class="QLabel" name="label_21">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>x</string>
          </property>
         </widget>
        </item>
        <item row="9" column="1">
         <widget class="QLabel" name="label_22">
          <property name="text">
           <string>y</string>
          </property>
         </widget>
        </item>
        <item row="10" column="2">
         <widget class="QLineEdit" name="modi_ang">
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QLabel" name="label_16">
          <property name="text">
           <string>state:</string>
          </property>
         </widget>
        </item>
        <item row="7" column="1">
         <widget class="QLineEdit" name="load_y">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item row="6" column="0">
         <widget class="QLabel" name="label_17">
          <property name="text">
           <string>loadx:</string>
          </property>
         </widget>
        </item>
        <item row="7" column="0">
         <widget class="QLabel" name="label_18">
          <property name="text">
           <string>loady:</string>
          </property>
         </widget>
        </item>
        <item row="6" column="1">
         <widget class="QLineEdit" name="load_x">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QPushButton" name="auto_save">
          <property name="text">
           <string>auto save</string>
          </property>
         </widget>
        </item>
        <item row="11" column="2">
         <widget class="QPushButton" name="modi_pos">
          <property name="text">
           <string>modi pos</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>底盘</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="1" column="2">
         <widget class="QPushButton" name="speedmax_bt">
          <property name="text">
           <string>改</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QLineEdit" name="anglespeed"/>
        </item>
        <item row="4" column="2">
         <widget class="QPushButton" name="moveradium_bt">
          <property name="text">
           <string>改</string>
          </property>
         </widget>
        </item>
        <item row="5" column="2">
         <widget class="QPushButton" name="angleradium_bt">
          <property name="text">
           <string>改</string>
          </property>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QLineEdit" name="angleradium"/>
        </item>
        <item row="4" column="1">
         <widget class="QLineEdit" name="moveradium"/>
        </item>
        <item row="7" column="1">
         <widget class="QLineEdit" name="factor"/>
        </item>
        <item row="6" column="1">
         <widget class="QLineEdit" name="start_2"/>
        </item>
        <item row="9" column="1">
         <widget class="QLineEdit" name="speedmin"/>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="speedmax"/>
        </item>
        <item row="0" column="0">
         <widget class="QPushButton" name="chassis_read">
          <property name="text">
           <string>读取参数</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>加减速比例</string>
          </property>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>旋转停范围</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>旋转速度比例</string>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>停车范围</string>
          </property>
         </widget>
        </item>
        <item row="6" column="0">
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>起始距离</string>
          </property>
         </widget>
        </item>
        <item row="8" column="0">
         <widget class="QLabel" name="label_11">
          <property name="text">
           <string>减速x系数</string>
          </property>
         </widget>
        </item>
        <item row="7" column="0">
         <widget class="QLabel" name="label_10">
          <property name="text">
           <string>factor</string>
          </property>
         </widget>
        </item>
        <item row="9" column="0">
         <widget class="QLabel" name="label_12">
          <property name="text">
           <string>最小速度</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLineEdit" name="movespeed"/>
        </item>
        <item row="3" column="2">
         <widget class="QPushButton" name="anglespeed_bt">
          <property name="text">
           <string>改</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QPushButton" name="chassis_save">
          <property name="text">
           <string>保存参数</string>
          </property>
         </widget>
        </item>
        <item row="8" column="2">
         <widget class="QPushButton" name="xfactor_bt">
          <property name="text">
           <string>改</string>
          </property>
         </widget>
        </item>
        <item row="7" column="2">
         <widget class="QPushButton" name="factor_bt">
          <property name="text">
           <string>改</string>
          </property>
         </widget>
        </item>
        <item row="6" column="2">
         <widget class="QPushButton" name="start_bt">
          <property name="text">
           <string>改</string>
          </property>
         </widget>
        </item>
        <item row="8" column="1">
         <widget class="QLineEdit" name="xfactor"/>
        </item>
        <item row="2" column="2">
         <widget class="QPushButton" name="movespeed_bt">
          <property name="text">
           <string>改</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>最大速度</string>
          </property>
         </widget>
        </item>
        <item row="9" column="2">
         <widget class="QPushButton" name="speedmin_bt">
          <property name="text">
           <string>改</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item row="5" column="6">
     <widget class="QLabel" name="ang">
      <property name="text">
       <string>ang:</string>
      </property>
     </widget>
    </item>
    <item row="3" column="6">
     <widget class="QLabel" name="x">
      <property name="text">
       <string>x:</string>
      </property>
     </widget>
    </item>
    <item row="8" column="6">
     <widget class="QLabel" name="label_20">
      <property name="styleSheet">
       <string notr="true">font: 18pt &quot;Adobe Arabic&quot;;</string>
      </property>
      <property name="text">
       <string>     ang:</string>
      </property>
     </widget>
    </item>
    <item row="7" column="8">
     <widget class="QLineEdit" name="pos_y">
      <property name="text">
       <string>0</string>
      </property>
     </widget>
    </item>
    <item row="4" column="8">
     <widget class="QPushButton" name="toLoad">
      <property name="text">
       <string>to_load</string>
      </property>
     </widget>
    </item>
    <item row="16" column="0">
     <widget class="QPushButton" name="start">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>start</string>
      </property>
     </widget>
    </item>
    <item row="16" column="1" colspan="8">
     <widget class="QPushButton" name="stop">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>stop</string>
      </property>
      <property name="shortcut">
       <string>Ctrl+Z</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
